File: lib/models/tv_channel.dart
class TvChannel {
  const TvChannel({
    required this.id,
    required this.image,
    required this.name,
    required this.resolutions,
    required this.paymentType,
    required this.hasAccess,
  });

  final String id;
  final String image;
  final String name;
  final Map<String, String> resolutions;
  final String paymentType;
  final bool hasAccess;

  Map<String, dynamic> toJson() {
    final map = <String, dynamic>{};
    map['id'] = id;
    map['image'] = image;
    map['name'] = name;
    map['resolutions'] = resolutions;
    map['paymentType'] = paymentType;
    map['hacAccess'] = hasAccess;
    return map;
  }
}

File: android/src/main/kotlin/uz/udevs/udevs_video_player/models/configuration/TvChannel.kt
package uz.udevs.udevs_video_player.models.configuration

import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class TvChannel(
    @SerializedName("id")
    val id: String,
    @SerializedName("image")
    val image: String,
    @SerializedName("name")
    val name: String,
    @SerializedName("resolutions")
    var resolutions: HashMap<String, String>,
    @SerializedName("paymentType")
    var paymentType: String,
    @SerializedName("hasAccess")
    var hasAccess: Boolean,
) : Serializable